File: Strings/GroupAnagrams.cpp
#include <vector>
#include <unordered_map>

using namespace std;
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> m;
        vector<vector<string> > res;
        for(string s : strs){
            string hash = s;
            sort(hash.begin(),hash.end());
            if(m.find(hash)==m.end()){
                m[hash]= vector<string>();
            }
            m[hash].push_back(s);
        }
        for(auto &p:m){
            res.push_back(p.second);
        }
        return res;
        
    }
};

File: Strings/GroupAnagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        hm = {}
        for x in strs:
            sorted_x = ''.join(sorted(x))
            hm[sorted_x] = hm.get(sorted_x,[])
            hm[sorted_x].append(x)
        return hm.values()